asserts/database.go

```diff
@@ -29,29 +29,21 @@ import (
 	"time"
 )
 
-// BuilderFromComps can build an assertion from its components.
-type BuilderFromComps func(headers map[string]string, body, content, signature []byte) (Assertion, error)
-
 // Backstore is a backstore for assertions. It can store and retrieve
-// assertions by type under primary paths (tuples of strings). Plus it
-// supports more general searches.
+// assertions by type under unique primary keys. Plus it
+// supports searching by headers.
 type Backstore interface {
-	// Init initializes the backstore. It is provided with a function
-	// to build assertions from their components.
-	Init(buildAssert BuilderFromComps) error
-	// Put stores an assertion under the given unique primaryPath.
+	// Put stores an assertion for the given primaryKey field names forming a unique key.
 	// It is responsible for checking that assert is newer than a
 	// previously stored revision.
-	Put(assertType AssertionType, primaryPath []string, assert Assertion) error
-	// Get loads an assertion with the given unique primaryPath.
+	Put(assertType AssertionType, primaryKey []string, assert Assertion) error
+	// Get loads the assertion with the given unique keyValues for the primaryKey fields.
 	// If none is present it returns ErrNotFound.
-	Get(assertType AssertionType, primaryPath []string) (Assertion, error)
+	Get(assertType AssertionType, primaryKey, keyValues []string) (Assertion, error)
```

Comment:
I don't find `primaryKey` totally obvious. Maybe s/primaryKey/keyHeaders and s/keyValues/key ?